--- src/boltz/data/write/mmcif.py
+++ src/boltz/data/write/mmcif.py
@@ -10,6 +10,8 @@
 from boltz.data import const
 from boltz.data.types import Structure

+RE_DIGIT = re.compile(r"\d")
+
 def to_mmcif(
     structure: Structure,
     plddts: Optional[Tensor] = None,
@@ -166,7 +168,7 @@

                         if boltz2:
                             atom_name = str(atom["name"])
-                            atom_key = re.sub(r"\d", "", atom_name)
+                            atom_key = RE_DIGIT.sub("", atom_name)
                             if atom_key in const.ambiguous_atoms:
                                 if isinstance(const.ambiguous_atoms[atom_key], str):
                                     element = const.ambiguous_atoms[atom_key]
